src/Ardalis.ApiEndpoints/CodeAnalyzers/EndpointHasExtraPublicMethodAnalyzer.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Ardalis.ApiEndpoints.CodeAnalyzers
+{
+    /// <summary>
+    /// TODO - update readme on how to manually verify: dotnet pack, open new VS, set Nuget repo to local directory
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class EndpointHasExtraPublicMethodAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "ApiEndpoints1001";
+
+        internal static readonly LocalizableString Title = "Endpoint has extra public method";
+        internal static readonly LocalizableString MessageFormat = "Endpoint {0} has an extra public method {1}";
+        private static readonly LocalizableString Description = "MVC will interpret additional public methods on an Endpoint as Actions.  Limit Endpoints to a single Action";
```

Comment:
@ardalis  - thoughts on what the Title/Message/Description should say?

Reply:
Title: Endpoint has more than one public method.
MessageFormat: "Endpoint {0} has additional public method {1}. Endpoints must have only one public method."
Description: sounds good as you have it

Reply:
![image](https://user-images.githubusercontent.com/1190907/87765660-ece66780-c7cc-11ea-826f-9695b463e9ac.png)